src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -427,6 +447,91 @@ OpenTelemetryPlugin::OpenTelemetryPlugin(
       });
 }
 
+OpenTelemetryPlugin::CallbackMetricReporter::CallbackMetricReporter(
```

Comment:
Please move the code for `CallbackMetricReporter` to before the code for `OpenTelemetryPlugin` itself (line 243), so that each class' methods are all defined in the same place.